apps/web/app/api/v1/js/lib/sync.ts

```diff
@@ -0,0 +1,181 @@
+import { getSyncSurveysCached } from "@/app/api/v1/js/lib/surveys";
+import { MAU_LIMIT } from "@formbricks/lib/constants";
+import { getActionClasses } from "@formbricks/lib/actionClass/service";
+import { getEnvironment } from "@formbricks/lib/environment/service";
+import { getMonthlyActivePeopleCount, getPerson } from "@formbricks/lib/person/service";
+import { getProductByEnvironmentId } from "@formbricks/lib/product/service";
+// import { createSession, getSession } from "@formbricks/lib/session/service";
+import { captureTelemetry } from "@formbricks/lib/telemetry";
+import { TEnvironment } from "@formbricks/types/environment";
+import { TJsState } from "@formbricks/types/js";
+// import { TSession } from "@formbricks/types/sessions";
+import { getSurveys } from "@formbricks/lib/survey/service";
+
+const captureNewSessionTelemetry = async (jsVersion?: string): Promise<void> => {
+  await captureTelemetry("state update", { jsVersion: jsVersion ?? "unknown" });
+};
+
+export const getUpdatedState = async (
+  environmentId: string,
+  personId: string,
+  jsVersion?: string
+): Promise<TJsState> => {
+  let environment: TEnvironment | null;
+  // let person: TPerson | null = null;
+  // let session: TSession | null;
+
+  if (jsVersion) {
+    captureNewSessionTelemetry(jsVersion);
+  }
+
+  // check if environment exists
+  environment = await getEnvironment(environmentId);
+
+  if (!environment) {
+    throw new Error("Environment does not exist");
+  }
+
+  // check if Monthly Active Users limit is reached
+  const currentMau = await getMonthlyActivePeopleCount(environmentId);
+  const isMauLimitReached = currentMau >= MAU_LIMIT;
+
+  if (isMauLimitReached) {
```

Comment:
please check current main branch for the updated version of this code to work with the specific team limits based on the plan of the team